pkg/client/client.go

```diff
@@ -230,128 +239,111 @@ Response body: %s`, endpoint, jsonToSend, resp.StatusCode, respBytes)
+func (c *Client) DeleteDocument(vaultID, docID string, opts ...ReqOption) error {
+	reqOpt := &ReqOpts{}
 
-	resp, err := c.httpClient.Post(fullEndpoint, "application/json", //nolint: bodyclose
-		bytes.NewBuffer(jsonToSend))
-	if err != nil {
-		return fmt.Errorf("failed to send POST request: %w", err)
+	for _, o := range opts {
+		o(reqOpt)
 	}
 
-	defer closeReadCloser(resp.Body)
+	endpoint := c.edvServerURL + fmt.Sprintf("/%s/documents/%s", url.PathEscape(vaultID), url.PathEscape(docID))
 
-	respBytes, err := ioutil.ReadAll(resp.Body)
+	statusCode, _, respBytes, err := c.sendHTTPRequest(
+		http.MethodDelete, endpoint, nil, c.getHeaderFunc(reqOpt))
 	if err != nil {
-		return fmt.Errorf("failed to read response body: %w", err)
+		return err
 	}
 
-	logger.Debugf(`Sent POST request to %s.
-Request body: %s
-
-Response status code: %d
-Response body: %s`, fullEndpoint, jsonToSend, resp.StatusCode, respBytes)
-
-	if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNoContent {
+	if statusCode == http.StatusOK {
 		return nil
 	}
 
 	return fmt.Errorf("the EDV server returned status code %d along with the following message: %s",
-		resp.StatusCode, respBytes)
+		statusCode, respBytes)
 }
 
-// DeleteDocument sends the EDV server a request to delete the specified document.
-func (c *Client) DeleteDocument(vaultID, docID string) error {
-	return c.sendDELETERequest(fmt.Sprintf("/%s/documents/%s", url.PathEscape(vaultID), url.PathEscape(docID)))
-}
+func (c *Client) sendHTTPRequest(method, endpoint string, body []byte,
+	addHeadersFunc addHeaders) (int, http.Header, []byte, error) {
+	req, errReq := http.NewRequest(method, endpoint, bytes.NewBuffer(body))
+	if errReq != nil {
+		return -1, nil, nil, errReq
+	}
 
-func (c *Client) sendDELETERequest(endpointPathToAppend string) error {
-	fullEndpoint := c.edvServerURL + endpointPathToAppend
+	if addHeadersFunc != nil {
+		httpHeaders, err := addHeadersFunc(req)
+		if err != nil {
+			return -1, nil, nil, fmt.Errorf("add optional request headers error: %w", err)
+		}
 
-	req, err := http.NewRequest(http.MethodDelete, fullEndpoint, nil)
-	if err != nil {
-		return fmt.Errorf("failed to create new DELETE request: %w", err)
+		if httpHeaders != nil {
+			req.Header = httpHeaders.Clone()
+		}
+	}
+
+	if method == http.MethodPost {
+		req.Header.Set("Content-Type", "application/json")
 	}
 
 	resp, err := c.httpClient.Do(req) //nolint: bodyclose
 	if err != nil {
-		return fmt.Errorf("failed to send DELETE request: %w", err)
+		return -1, nil, nil, err
 	}
 
 	defer closeReadCloser(resp.Body)
 
 	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return fmt.Errorf("failed to read response body: %w", err)
+		return -1, nil, nil, err
 	}
 
-	logger.Debugf(`Sent DELETE request to %s.
-Response status code: %d
-Response body: %s`, fullEndpoint, resp.StatusCode, respBytes)
+	logger.Debugf(`sent %s request to %s response status code: %d response body: %s`, method, endpoint,
```

Comment:
Any reason for the removal of the new lines in the debug message? I'm afraid that it's going to be very difficult to read this log output.

Reply:
@DRK3 ops don't like multi line logs, it makes it difficult searching/troubleshooting entries

Reply:
feedback from dev ops: new lines in debug is nightmare

Reply:
OK! I'll avoid them from now on.